### PR TITLE
Implemented POST /api/questionbank/import/preview endpoint (#41)

### LINKED ISSUE #41: POST /api/sessions/{sessionId}/questions/import — preview parse endpoint
## Description

Accepts multipart file and returns parsed rows with validation messages.

## Acceptance Criteria

- [x] Returns parsed preview and per-row validation results

Implementation Checklist

- [x] Create a CsvPreviewEndpointHandlers.cs with a POST /api/questionbank/import/preview endpoint handler.
- [x] Accept a multipart form file upload (IFormFile).
- [x] Validate the file is present and has a .csv extension — return 400 if not.
- [x] Parse the CSV file using the existing CSV parser from #40 — check how it was implemented first.
- [x] For each parsed row, validate:
- [x] Text is present and non-empty
- [x] Type is a valid QuestionType value
- [x] If Type is MultipleChoice, at least 2 options are present
- [x] Build a preview response containing each row with its parsed fields and a list of validation messages per row.
- [x] Return 200 with the preview result even if some rows have validation errors — the caller decides whether to proceed.
- [x] Register the route in Program.cs.
- [x] Add unit tests covering: valid CSV, CSV with some invalid rows, empty file, missing file, wrong file type.

Definition of Done

- [x] Endpoint implemented and reachable at POST /api/questionbank/import/preview.
- [x] Returns 200 with parsed rows and per-row validation messages.
- [x] Returns 400 for missing file or wrong file type.
- [x] Validation checks text, type, and MC option count per row.
- [x] Invalid rows are included in the response with their validation messages — not rejected outright.
- [x] Unit tests added and passing.
- [x] dotnet build succeeds.
- [x] Linting passes (dotnet format --verify-no-changes).

### PR DESCRIPTION
**Pull request by ljkaminski3**
## Description
Implements the `POST /api/questionbank/import/preview` endpoint that accepts a CSV file upload and returns parsed rows with per-row validation results.

- Added `CsvPreviewEndpointHandlers` with `PreviewImport` handler
- Added `CsvQuestionRow` model with `Text`, `Type`, `Options`, `ValidationErrors`, and `IsValid` fields
- Added `CsvPreviewResponse` record with `Rows` and `AllRowsValid`
- Uses the existing `CsvParser` from #40 for parsing
- CSV format: `text,type,options` where options are pipe-separated (e.g. `A|B|C`)
- Validates text is present, type is a valid `QuestionType`, and MC questions have at least 2 options
- Invalid rows are included in the response with their validation errors — not rejected outright
- Returns 200 with all rows (valid and invalid) and `allRowsValid` flag on success
- Returns 400 for missing file, empty file, or non-CSV file type
- Registered route at `POST /api/questionbank/import/preview` with antiforgery disabled for file upload
- Unit tests added covering all scenarios — all passing

## Related Issues
Closes #41

## Checklist
- [x] Tests added/updated
- [x] Documentation updated
- [x] Linting passes

Screenshot 2026-04-23 at 3 06 07 PM
Screenshot 2026-04-23 at 3 05 59 PM
Screenshot 2026-04-23 at 2 59 31 PM

### CODE REVIEW

**Comment by ljkaminski3** [comment]
@copilot do not make any changes. please conduct a small review on this PR and search for any hidden bugs. Requirements below:

Description
Accepts multipart file and returns parsed rows with validation messages.

Acceptance Criteria

Returns parsed preview and per-row validation results
Implementation Checklist


Create a CsvPreviewEndpointHandlers.cs with a POST /api/questionbank/import/preview endpoint handler.

Accept a multipart form file upload (IFormFile).

Validate the file is present and has a .csv extension — return 400 if not.

Parse the CSV file using the existing CSV parser from https://github.com/ECU-Pirate-Forge/pulse-2.0/issues/40 — check how it was implemented first.

For each parsed row, validate:

Text is present and non-empty

Type is a valid QuestionType value

If Type is MultipleChoice, at least 2 options are present

Build a preview response containing each row with its parsed fields and a list of validation messages per row.

Return 200 with the preview result even if some rows have validation errors — the caller decides whether to proceed.

Register the route in Program.cs.

Add unit tests covering: valid CSV, CSV with some invalid rows, empty file, missing file, wrong file type.
Definition of Done


Endpoint implemented and reachable at POST /api/questionbank/import/preview.

Returns 200 with parsed rows and per-row validation messages.

Returns 400 for missing file or wrong file type.

Validation checks text, type, and MC option count per row.

Invalid rows are included in the response with their validation messages — not rejected outright.

Unit tests added and passing.

dotnet build succeeds.

Linting passes (dotnet format --verify-no-changes).

**Comment by Mtrinh123** [comment]
@copilot no changes, Please make sure everything is running with no errors and hidden bugs!

File: src/Pulse.WebApi/CsvPreviewEndpointHandlers.cs [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+                errors.Add("Text is required.");
+
+            if (!int.TryParse(typeRaw, out var typeInt) || !Enum.IsDefined(typeof(QuestionType), typeInt))
+                errors.Add($"Type '{typeRaw}' is not a valid question type (0=MultipleChoice, 1=LikertScale, 2=OpenEnded).");
+
+            var options = string.IsNullOrWhiteSpace(optionsRaw)
+                ? new List<string>()
+                : optionsRaw.Split('|').Select(o => o.Trim()).Where(o => !string.IsNullOrWhiteSpace(o)).ToList();
+
+            if (Enum.IsDefined(typeof(QuestionType), typeInt) && (QuestionType)typeInt == QuestionType.MultipleChoice && options.Count < 2)
```
Comment (Mtrinh123) [inline src/Pulse.WebApi/CsvPreviewEndpointHandlers.cs]: When typeRaw cannot be parsed as an integer, typeInt silently defaults to 0 (C# out parameter default). Because 0 == QuestionType.MultipleChoice, the subsequent MC-options guard incorrectly fires, producing a spurious "Multiple choice questions require at least 2 options" error on top of the "not a valid question type" error. The fix is to gate the MC check on a typeIsValid bool rather than re-checking Enum.IsDefined on the defaulted value. No test currently exercises this path, which is why all 133 tests pass despite the bug being present.

(Hidden Bug Fix)

**Review by Mtrinh123 — CHANGES_REQUESTED** [review]
There's one hidden bug you need to change, afterward, everything will be fine to merge!

**Comment by ljkaminski3** [comment]
@copilot do not make any changes. check the most recent commit for hidden bugs and check all previous requirements pass

**Review by Mtrinh123 — APPROVED** [review]
I see the hidden bug has been fixed and everything is up to date! You can merge whenever. Good Stuff!
